services/net/reporting/ReportingManager.cs

```diff
@@ -351,9 +354,19 @@ private async Task GenerateReportAsync(ReportRequestModel request, API.Areas.Ser
             }
         }
 
-        var to = report.Subscribers.Where(s => !String.IsNullOrWhiteSpace(s.User?.Email) && s.IsSubscribed).Select(s => s.User!.Email).ToArray();
-        var subject = await this.ReportEngine.GenerateReportSubjectAsync(report, sectionContent, false);
-        var body = await this.ReportEngine.GenerateReportBodyAsync(report, sectionContent, null, false);
+        // Generate and send report to subscribers who want an email with a link to the website.
+        // We do this first because we don't want to save the output of this in the instance.
+        var linkOnlyFormatSubscribers = report.Subscribers.Where(s => s.IsSubscribed && LinkOnlyFormats.Contains(s.Format) && !String.IsNullOrWhiteSpace(s.User?.Email)).ToArray();
+        var linkOnlyFormatBody = linkOnlyFormatSubscribers.Any() ? await this.ReportEngine.GenerateReportBodyAsync(report, sectionContent, null, true, false) : "";
```

Comment:
running them in a specific order to avoid a problem seems a little fragile to me.  Would it be better to have a mechanism for creating the reportbody without saving the instance at all?

Reply:
or should different format mean different/seperate instance for each?

Reply:
I was going back and forth on this and trying to think of other possibilities.  I'm not a fan of the two separate email groups as part of a single operation, but it seemed the best option in this case.

- We only want a single instance of a report when it is published
- When we generate an instance, we want all reports to reference it so that when a user goes to the website and views it, it'll be already generated and identical each time
- The link only email is really the same instance of the report and must point to the instance with content

Reply:
Originally I designed it so that it was a different report or instance to handle the different format.  This however didn't meet the business requirements.